pkg/build/controller/build/build_controller.go

```diff
@@ -1367,6 +1389,52 @@ func (bc *BuildController) handleBuildConfigError(err error, key interface{}) {
 	bc.buildConfigQueue.Forget(key)
 }
 
+// createBuildCAConfigMap creates a ConfigMap containing certificate authorities used by the build pod.
+func (bc *BuildController) createBuildCAConfigMap(build *buildv1.Build, buildPod *corev1.Pod, update *buildUpdate) (*buildUpdate, error) {
+	configMapSpec := bc.createBuildCAConfigMapSpec(build, buildPod)
+	configMap, err := bc.configMapClient.ConfigMaps(buildPod.Namespace).Create(configMapSpec)
+	if err != nil {
+		bc.recorder.Eventf(build, corev1.EventTypeWarning, "FailedCreate", "Error creating build certificate authority: %v", err)
+		update.setReason("CannotCreateCAConfigMap")
+		update.setMessage(buildutil.StatusMessageCannotCreateCAConfigMap)
+		return update, fmt.Errorf("failed to create build certificate authority: %v", err)
+	}
+	glog.V(4).Infof("Created certificate authority configMap %s/%s for build %s", build.Namespace, configMap.Name, buildDesc(build))
+	return update, nil
+}
+
+// createBuildCAConfigMapSpec creates a ConfigMap template to hold certificate authorities
+// used by the build pod.
+// The returned ConfigMap has an owner reference to the provided pod, ensuring proper
+// garbage collection.
+func (bc *BuildController) createBuildCAConfigMapSpec(build *buildv1.Build, buildPod *corev1.Pod) *corev1.ConfigMap {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
you might as well put the service-ca annotation on here now.  it won't hurt anything, just don't explicitly try to mount the expected service-ca key.